### ISSUE TITLE
[Feature] Reset value type check

### ISSUE THREAD

**Issue by taichi-ishitani** [veryl-lang/veryl#584]
In most cases, a reset value should be a constant value (parameter, localparam, literal).
I think it's useful that the linter reports warnings if a non-constant value is used as a reset value.

```systemverilog
var a: logic;
var b: logic;
always_ff (i_clk, i_rst_n) {
  if_reset {
    b = a; // report warning
  }
}
```

**Comment by nblei** [comment]
@dalance Is there currently a way to extract a `TokenRange` or `SourceSpan` from an `Expression`?

**Comment by nblei** [comment]
@dalance This can be closed by #751
